From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Sat, 18 Jul 2026 23:59:19 +0200
Subject: refactor(crypt): use function in crypt-cleanup.sh

shellcheck will complain about crypt-cleanup.sh when checking for
busybox:

```
In modules.d/70crypt/crypt-cleanup.sh line 8:
        local do_break="y"
        ^---^ SC2168 (error): 'local' is only valid in functions.
```

Origin: upstream, https://github.com/dracut-ng/dracut/commit/36fe74ff0b51afc83c223c7e6d4371152ce63f86
---
 modules.d/70crypt/crypt-cleanup.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/modules.d/70crypt/crypt-cleanup.sh b/modules.d/70crypt/crypt-cleanup.sh
index 0539e70..7d03763 100755
--- a/modules.d/70crypt/crypt-cleanup.sh
+++ b/modules.d/70crypt/crypt-cleanup.sh
@@ -3,12 +3,16 @@
 # close everything which is not busy
 rm -f -- /etc/udev/rules.d/70-luks.rules > /dev/null 2>&1
 
-if ! getarg rd.luks.uuid > /dev/null 2>&1 && getargbool 1 rd.luks > /dev/null 2>&1; then
-    while true; do
+close_luks() {
+    while :; do
         local do_break="y"
         for i in /dev/mapper/luks-*; do
             cryptsetup luksClose "$i" > /dev/null 2>&1 && do_break=n
         done
         [ "$do_break" = "y" ] && break
     done
+}
+
+if ! getarg rd.luks.uuid > /dev/null 2>&1 && getargbool 1 rd.luks > /dev/null 2>&1; then
+    close_luks
 fi
